File: js_checkio_solutions/Home/non_unique_elements.js
#!/usr/bin/env checkio --domain=js run non-unique-elements
"use strict";
// If you have 50 different plug types, appliances wouldn't be available and would be very        expensive. But once an electric outlet becomes standardized, many companies can design        appliances, and competition ensues, creating variety and better prices for consumers.
// -- Bill Gates
// 
// You are given a non-empty list of integers (X).    For this task, you should return a list consisting of only the non-unique elements in this list.    To do so you will need to remove all unique elements (elements which are contained in a given    list only once).    When solving this task, do not change the order of the list.    Example: [1, 2, 3, 1, 3] 1 and 3 non-unique elements and result will be [1, 3, 1, 3].
// 
// 
// 
// Input:A list of integers.
// 
// Output:An iterable of integers.
// 
// 
// How it is used:This mission will help you to understand how to manipulate arrays,    something that is the basis for solving more complex tasks.    The concept can be easily generalized for real world tasks.    For example: if you need to clarify statistics by removing low frequency elements (noise).
// 
// You can find out more about JavaScript arrays inone of our articles featuring lists, tuples, array.array and numpy.array.
// 
// Precondition:
// 0 < len(data) < 1000
// 
// 
// 
// END_DESC
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var assert_1 = __importDefault(require("assert"));
function nonUniqueElements(data) {
    return [];
}
console.log('Example:');
console.log(nonUniqueElements([1, 2, 3, 1, 3]));
assert_1.default.deepEqual(nonUniqueElements([1, 2, 3, 1, 3]), [1, 3, 1, 3]);
assert_1.default.deepEqual(nonUniqueElements([1, 2, 3, 4, 5]), []);
assert_1.default.deepEqual(nonUniqueElements([5, 5, 5, 5, 5]), [5, 5, 5, 5, 5]);
assert_1.default.deepEqual(nonUniqueElements([10, 9, 10, 10, 9, 8]), [10, 9, 10, 10, 9]);
console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");

File: py_checkio_solutions/Home/non_unique_elements.py
#!/usr/bin/env checkio --domain=py run non-unique-elements

# If you have 50 different plug types, appliances wouldn't be available and would be very        expensive. But once an electric outlet becomes standardized, many companies can design        appliances, and competition ensues, creating variety and better prices for consumers.
# -- Bill Gates
# 
# You are given a non-empty list of integers (X).    For this task, you should return a list consisting of only the non-unique elements in this list.    To do so you will need to remove all unique elements (elements which are contained in a given    list only once).    When solving this task, do not change the order of the list.    Example: [1, 2, 3, 1, 3] 1 and 3 non-unique elements and result will be [1, 3, 1, 3].
# 
# 
# 
# Input:A list of integers.
# 
# Output:An iterable of integers.
# 
# 
# How it is used:This mission will help you to understand how to manipulate arrays,    something that is the basis for solving more complex tasks.    The concept can be easily generalized for real world tasks.    For example: if you need to clarify statistics by removing low frequency elements (noise).
# 
# You can find out more about Python arrays inone of our articles featuring lists, tuples, array.array and numpy.array.
# 
# Precondition:
# 0 < len(data) < 1000
# 
# 
# 
# END_DESC

#Your optional code here
#You can import some modules or create additional functions


def checkio(data: list) -> list:
    #Your code here
    #It's main function. Don't remove this function
    #It's used for auto-testing and must return a result for check.  
    s = set(data)
    d = {f: 0 for f in s}
    for x in data:
        if x in d:
            d[x] += 1

    r = set()

    for k, v in d.items():
        if v == 1:
            r.add(k)

    for x in r:
        data.remove(x)

    # replace this for solution
    return data

#Some hints
#You can use list.count(element) method for counting.
#Create new list with non-unique elements
#Loop over original list


if __name__ == "__main__":
    #These "asserts" using only for self-checking and not necessary for auto-testing
    assert list(checkio([1, 2, 3, 1, 3])) == [1, 3, 1, 3], "1st example"
    assert list(checkio([1, 2, 3, 4, 5])) == [], "2nd example"
    assert list(checkio([5, 5, 5, 5, 5])) == [5, 5, 5, 5, 5], "3rd example"
    assert list(checkio([10, 9, 10, 10, 9, 8])) == [10, 9, 10, 10, 9], "4th example"
    print("It is all good. Let's check it now")